src/rgw/rgw_sal_filter.cc

```diff
@@ -1044,6 +1044,17 @@ RGWAccessControlPolicy& FilterObject::get_acl()
   return next->get_acl();
 }
 
+int FilterObject::list_parts(const DoutPrefixProvider* dpp, CephContext* cct,
+			     int max_parts, int marker, int* next_marker,
+			     bool* truncated, list_parts_each_t each_func,
+			     optional_yield y)
+{
+  return next->list_parts(dpp, cct, max_parts, marker, next_marker,
+			  truncated,
+			  sal::Object::list_parts_each_t(each_func),
```

Comment:
does `std::move(each_func)` work here instead? i'm surprised that unique_function is copyable this way, especially if it lets you pass a lambda that captures something like a unique_ptr that isn't copyable

Reply:
I don't understand the problem here;  @adamemerson?